### ISSUE TITLE
Expressions like Tuple{Int64, String}([1,"5"]) are typeunstable

### ISSUE THREAD

**Issue by JTaets** [JuliaLang/julia#49712]
``` julia
@code_warntype Tuple{Int64, String}([1,"5"]) #Tuple{Any, String}
```
For longer expressions, it still is only unstable in the first argument.

(Note, I haven't upgraded yet to 1.9)
``` julia
Julia Version 1.8.5
Commit 17cfb8e65e (2023-01-08 06:45 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: 8 × Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-13.0.1 (ORCJIT, skylake)
  Threads: 1 on 8 virtual cores
Environment:
  JULIA_EDITOR = code
  JULIA_NUM_THREADS =
```

**Comment by bvdmitri** [comment]
I can reproduce on MacOS 1.8.5.
It is inferred correctly on 1.9 though: 

```julia
julia> @code_warntype Tuple{Int64, String}([1,"5"]) 
MethodInstance for Tuple{Int64, String}(::Vector{Any})
  from (::Type{T})(itr) where T<:Tuple @ Base tuple.jl:369
Static Parameters
  T = Tuple{Int64, String}
Arguments
  #self#::Core.Const(Tuple{Int64, String})
  itr::Vector{Any}
Body::Tuple{Int64, String}
1 ─ %1 = Base._totuple($(Expr(:static_parameter, 1)), itr)::Tuple{Int64, String}
└──      return %1
```

**Comment by KristofferC** [comment]
> Hence, it cannot be considered "already fixed", unless it's decided that it's not worth backporting the fix to 1.6.

Hm, it's not like we keep every issue open until it has been backported to 1.6. And this isn't even a bug. So I will close this.
